survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/LocationPickerPart.kt

```diff
@@ -0,0 +1,268 @@
+import android.content.pm.PackageManager
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.MotionEvent
+import android.view.View
+import android.view.animation.BounceInterpolator
+import android.view.inputmethod.InputMethodManager
+import android.widget.*
+import androidx.core.app.ActivityCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.GoogleMapOptions
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.model.LatLng
+import com.quickbirdstudios.surveykit.R
+import com.quickbirdstudios.surveykit.SurveyTheme
+import com.quickbirdstudios.surveykit.backend.address.AddressSuggestionProvider
+import com.quickbirdstudios.surveykit.backend.views.main_parts.StyleablePart
+
+class LocationPickerPart @JvmOverloads constructor(
+    context: Context,
+    private val lifecycle: Lifecycle,
+    addressProvider: AddressSuggestionProvider,
+    attrs: AttributeSet? = null
+) : LinearLayout(context, attrs), StyleablePart, LifecycleObserver {
+
+    //region Members
+    private var search: AutoCompleteTextView
+    private val mapView: MapView
+    private lateinit var map: GoogleMap
+    private val frame = FrameLayout(context)
+    private val locationMarker = ImageView(context)
+
+    private val latitude: Double?
+        get() {
+            return if (::map.isInitialized) {
+                map.cameraPosition.target.latitude
+            } else {
+                null
+            }
+        }
+
+    private val longitude: Double?
+        get() {
+            return if (::map.isInitialized) {
+                map.cameraPosition.target.longitude
+            } else {
+                null
+            }
+        }
+
+    private lateinit var mapReadySelection: Selection
+
+    //endregion
+
+    //region Public API
+
+    var selected: Selection
+        get() = Selection(
+            latitude, longitude
+        )
+        set(selected) {
+            if (::map.isInitialized && selected.longitude != null && selected.latitude != null) {
+                map.moveCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        LatLng(
+                            selected.longitude,
+                            selected.latitude
+                        ),
+                        15f
+                    )
+                )
+            } else {
+                mapReadySelection = selected
+            }
+        }
+
+    //endregion
+
+    //region Overrides
+
+    override fun style(surveyTheme: SurveyTheme) {}
+
+    //endregion
+
+    //region Subtypes
+
+    data class Selection(val latitude: Double?, val longitude: Double?)
+
+    //endregion
+
+    init {
```

Comment:
the `init` block should be in the beginning of the file

Reply:
As a convention, in this project, the `init` block should always be at the bottom of the file 🤷‍♂️